projects/chaos/inc/relay.h

```diff
@@ -0,0 +1,25 @@
+#pragma once
+
+#include <stdbool.h>
+
+#include "event_queue.h"
+
+// TODO(ELEC-105): Export this to a shared library for use by boards with Relays.
+typedef enum {
+  RELAY_ID_SOLAR_MASTER_FRONT = 0,
+  RELAY_ID_SOLAR_MASTER_REAR,
+  RELAY_ID_BATTERY,
+  RELAY_ID_MAIN_POWER,
+  NUM_RELAY_IDS,
+} RelayId;
+
+typedef enum {
+  RELAY_STATE_OPEN = 0,
+  RELAY_STATE_CLOSE = 1,
+} RelayState;
+
+// Initializes the relay FSMs.
+void relay_init(bool loopback);
```

Comment:
Suggest having the user pass in storage for the FSMs instead of allocating them statically

Reply:
I switched to a model-controller style the FSM is abstracted away to its own module now. I'm still happy with making these static with a thin wrapper. The main motivation is to not have ~10 FSMs sitting around in my main loop that feels gross to manage so I would prefer lightweight wrappers that provide the storage. I understand giving users control hence the refactor to the separate module but I still strongly prefer this style of interface in my main loop for Chaos.